source/common/formatter/substitution_formatter.cc

```diff
@@ -814,9 +822,13 @@ StreamInfoFormatter::StreamInfoFormatter(const std::string& field_name) {
                      : absl::make_optional<std::string>(upstream_cluster_name);
         });
   } else if (field_name == "UPSTREAM_LOCAL_ADDRESS") {
-    field_extractor_ =
-        StreamInfoAddressFieldExtractor::withPort([](const StreamInfo::StreamInfo& stream_info) {
-          return stream_info.upstreamLocalAddress();
+    field_extractor_ = StreamInfoAddressFieldExtractor::withPort(
+        [](const StreamInfo::StreamInfo& stream_info)
+            -> std::shared_ptr<const Envoy::Network::Address::Instance> {
+          if (stream_info.upstreamInfo().has_value()) {
+            return stream_info.upstreamInfo().value().get().upstreamLocalAddress();
```

Comment:
Nit: small simplification, since OptRef already has a method for this chain of calls.
```suggestion
            return stream_info.upstreamInfo().ref().upstreamLocalAddress();
```